superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { FunctionComponent, useState } from 'react';
+import styled from '@superset-ui/style';
+import { t } from '@superset-ui/translation';
+import withToasts from 'src/messageToasts/enhancers/withToasts';
+import Icon from 'src/components/Icon';
+import Modal from 'src/common/components/Modal';
+import { Tabs } from 'src/common/components';
+
+export type DatabaseObject = {
+  id: number;
+  name: string;
+  uri: string;
+  // TODO: add more props
+};
+
+interface DatabaseModalProps {
+  addDangerToast: (msg: string) => void;
+  addSuccessToast: (msg: string) => void;
+  onDatabaseAdd?: (database?: DatabaseObject) => void; // TODO: should we add a separate function for edit?
+  onHide: () => void;
+  show: boolean;
+  database?: DatabaseObject | null; // If included, will go into edit mode
+}
+
+const { TabPane } = Tabs;
+
+const StyledIcon = styled(Icon)`
+  margin: auto 10px auto 0;
+`;
+
+const StyledTabs = styled(Tabs)`
```

Comment:
If this styling to make tabs look more like the new redesign, I'd love to see this styled component get exported from `/common/...` so it can be widely re-used. That's on my to-do list anyway, so I volunteer to touch it up, add storybook entries, etc.

Reply:
I was considering that, yea. I can convert it!